core/procs.go

```diff
@@ -1388,14 +1389,44 @@ var procLoadFile Proc = func(args []Object) Object {
 	return loadFile(filename.S)
 }
 
-var procLoadLibFromFile Proc = func(args []Object) Object {
-	libname := EnsureSymbol(args, 0)
-	filename := EnsureString(args, 1)
-	if d := internalLibs[libname.Name()]; d != nil {
+var procLoadLibFromPath Proc = func(args []Object) Object {
+	libname := EnsureSymbol(args, 0).Name()
+	pathname := EnsureString(args, 1).S
+	if d := internalLibs[libname]; d != nil {
 		processData(d)
-	} else {
-		loadFile(filename.S)
+		return NIL
+	}
+	cp := GLOBAL_ENV.classPath.Value
+	cpvec := AssertVector(cp, "*classpath* must be a Vector, not a "+cp.GetType().ToString(false))
+	count := cpvec.Count()
+	var f *os.File
+	var err error
+	var canonicalErr error
+	var filename string
+	for i := 0; i < count; i++ {
+		elem := cpvec.at(i)
+		cpelem := AssertString(elem, "*classpath* must contain only Strings, not a "+elem.GetType().ToString(false)+" (at element "+strconv.Itoa(i)+")")
+		s := cpelem.S
+		if s == "" {
+			filename = pathname
+		} else {
+			filename = filepath.Join(s, filepath.Join(strings.Split(libname, ".")...)) + ".joke" // could cache inner join....
+		}
+		f, err = os.Open(filename)
+		if err == nil {
+			break
+		}
+		if s == "" {
+			canonicalErr = err
+		}
 	}
+	if canonicalErr == nil {
```

Comment:
Error handling here seems more complicated than it should be. What's the difference between `err` and `canonicalErr`? Don't we want to panic on `err` either way (since `canonicalErr` is either equal to `err` or `nil`)?

Reply:
OK, I think this could be simplified to 
```go
PanicOnErr(canonicalErr)
PanicOnErr(err)
```
However, I wonder if we should provide more user friendly error message here. Something like `Failed to load namespace X. Tried locations: f1, f2...`